Mimir/Models/Items/Item.cs

```diff
@@ -1,54 +1,125 @@
 using Nekoyume.Model.Item;
+using Nekoyume.Model.State;
 
 namespace Mimir.Models.Items;
 
 public class Item
 {
+    protected static readonly Codec Codec = new();
+
     public int ItemSheetId { get; set; }
     public int Grade { get; set; }
     public ItemType ItemType { get; set; }
     public ItemSubType ItemSubType { get; set; }
     public ElementalType ElementalType { get; set; }
-
     public int Count { get; set; }
 
-    public Guid? NonFungibleId { get; set; }
+    public int? Level { get; set; }
     public long? RequiredBlockIndex { get; set; }
+    public HashDigest<SHA256>? FungibleId { get; set; }
+    public Guid? NonFungibleId { get; set; }
+    public Guid? TradableId { get; set; }
 
-    public Item(ItemBase item, int count)
-    {
-        ItemSheetId = item.Id;
-        Grade = item.Grade;
-        ItemType = item.ItemType;
-        ItemSubType = item.ItemSubType;
-        ElementalType = item.ElementalType;
+    public Item(ItemBase itemBase, int count) => Reset(itemBase, count);
 
-        Count = count;
-
-        NonFungibleId = item is INonFungibleItem nonFungibleItem
-            ? nonFungibleItem.NonFungibleId
-            : null;
-        RequiredBlockIndex = item switch
-        {
-            INonFungibleItem nfi => nfi.RequiredBlockIndex,
-            ITradableItem ti => ti.RequiredBlockIndex,
-            _ => null
-        };
+    public Item(Inventory.Item inventoryItem) : this(inventoryItem.item, inventoryItem.count)
+    {
     }
 
-    public Item(BsonDocument item)
+    public Item(BsonDocument inventoryItem)
     {
-        ItemSheetId = item["ItemSheetId"].AsInt32;
+        var item = inventoryItem["item"].AsBsonDocument;
+        if (item.Contains("serialized"))
+        {
+            var serialized = item["serialized"].AsString;
+            var base64Encoded = Convert.FromBase64String(serialized);
+            var serializedDictionary = (Dictionary)Codec.Decode(base64Encoded);
+            var itemType = serializedDictionary["item_type"].ToEnum<ItemType>();
+            ItemBase itemBase = itemType switch
+            {
+                ItemType.Consumable => new Consumable(serializedDictionary),
+                ItemType.Costume => new Costume(serializedDictionary),
+                ItemType.Equipment => new Equipment(serializedDictionary),
+                ItemType.Material => new Material(serializedDictionary),
+                _ => throw new ArgumentOutOfRangeException(
+                    nameof(itemType),
+                    $"Invalid ItemType: {itemType}")
+            };
+            Reset(itemBase, inventoryItem["count"].AsInt32);
+            return;
+        }
+
+        ItemSheetId = item["Id"].AsInt32;
         Grade = item["Grade"].AsInt32;
         ItemType = (ItemType)item["ItemType"].AsInt32;
         ItemSubType = (ItemSubType)item["ItemSubType"].AsInt32;
         ElementalType = (ElementalType)item["ElementalType"].AsInt32;
+        Count = inventoryItem["count"].AsInt32;
```

Comment:
How about using `ToInt32`?

Reply:
원본 데이타의 타입이 `int`이어서 `AsInt32`를 사용하는 것이 좋을 것 같습니다.
그리고, 말씀해주신 덕에 "RequiredBlockIndex"의 변환 방법을 `ToInt64()`로 바꿀 수 있었습니다.(56a8d23)
I think it would be better to use `AsInt32` since the original data is of type `int`.([Nekoyume.Model.Item.Inventory.Item.count](https://github.com/planetarium/lib9c/blob/development/Lib9c/Model/Item/Inventory.cs#L29))
Also, thanks to your suggestion, I was able to change the conversion method of "RequiredBlockIndex" to `ToInt64()`.(56a8d23)